go/chat/boxer.go

```diff
@@ -61,14 +61,24 @@ type udCacheValue struct {
 	DeviceName string
 }
 
+func (b *Boxer) makeErrorMessage(msg chat1.MessageBoxed, err error) chat1.MessageFromServerOrError {
+	return chat1.MessageFromServerOrError{
+		UnboxingError: &chat1.MessageError{
+			Errmsg:      err.Error(),
+			MessageID:   msg.ServerHeader.MessageID,
+			MessageType: msg.ServerHeader.MessageType,
+		},
+	}
+}
+
 // unboxMessage unboxes a chat1.MessageBoxed into a keybase1.Message.  It finds
 // the appropriate keybase1.CryptKey.
 func (b *Boxer) UnboxMessage(ctx context.Context, finder *KeyFinder, boxed chat1.MessageBoxed) (res chat1.MessageFromServerOrError, err error) {
 	tlfName := boxed.ClientHeader.TlfName
 	tlfPublic := boxed.ClientHeader.TlfPublic
 	keys, err := finder.Find(ctx, b.tlf, tlfName, tlfPublic)
 	if err != nil {
-		return chat1.MessageFromServerOrError{}, libkb.ChatUnboxingError{Msg: err.Error()}
+		return b.makeErrorMessage(boxed, err), libkb.ChatUnboxingError{Msg: err.Error()}
```

Comment:
I made a ticket for improving this, but we should probably segment the types of unboxing errors. An error because we couldn't verify a key from the server or `LoadUser` fails is different from if the text doesn't decrypt. I'm just going to leave it be for this PR for now.